src/MeetingsSDKAdapter.js

```diff
@@ -0,0 +1,171 @@
+import {MeetingsAdapter} from '@webex/component-adapter-interfaces';
+import {Observable, fromEvent, concat} from 'rxjs';
+import {map, publishReplay, refCount, filter, finalize} from 'rxjs/operators';
+
+const LOCAL = 'local';
+const MEDIA_READY = 'media:ready';
+const MEDIA_STOPPED = 'media:stopped';
+
+/**
+ * The `MeetingsSDKAdapter` is an implementation of the `MeetingsAdapter` interface.
+ * This adapter utilizes the Webex JS SDK to create and join webex meetings.
+ *
+ * @class MeetingsSDKAdapter
+ * @extends {MeetingsAdapter}
+ */
+export default class MeetingsSDKAdapter extends MeetingsAdapter {
+  constructor(datasource) {
+    super(datasource);
+    this.meetingsObservables = {};
+  }
+
+  /**
+   * Register the SDK meeting plugin to the device
+   * and sync the meeting collection from the server.
+   */
+  async connect() {
+    await this.datasource.meetings.register();
+    await this.datasource.meetings.syncMeetings();
+  }
+
+  /**
+   * Unregister the sdk meeting plugin from the device.
+   */
+  async disconnect() {
+    await this.datasource.meetings.unregister();
+  }
+
+  /**
+   * Retrieve the local device media and add them with
+   * the given default media settings to the meeting.
+   *
+   * @param {Object} meeting  meeting instance to add the media to
+   */
+  async addLocalMedia(meeting) {
+    // default media setting
+    const mediaSettings = {
+      receiveVideo: true,
+      receiveAudio: true,
+      receiveShare: false,
+      sendVideo: true,
+      sendAudio: true,
+      sendShare: false,
+    };
+    const [localStream, localShare] = await meeting.getMediaStreams(mediaSettings);
```

Comment:
We won’t have access to getMediaStreams or addMedia on the meeting object our observable returns. 

We could do a getMeeting here or add this function to the observable object.

Reply:
taking it to the space